tests/host_tools/logging.py

```diff
@@ -15,23 +16,45 @@ class Fifo:
     path = None
     fifo = None
 
-    def __init__(self, path):
+    def __init__(self, path, blocking=False):
         """Create a new named pipe."""
         if os.path.exists(path):
             raise FileExistsError("Named pipe {} already exists.".format(path))
+
         os.mkfifo(path)
-        # Open the FIFO and set O_NONBLOCK flag.
-        fd = os.open(path, os.O_RDONLY | os.O_NONBLOCK)
-        self.fifo = os.fdopen(fd, "r")
+        if not blocking:
+            fd = os.open(path, os.O_NONBLOCK)
+            self.fifo = os.fdopen(fd, "r")
+        else:
+            self.fifo = open(path, "r")
+
         self.path = path
 
     def sequential_reader(self, max_lines):
-        """Return up to `max_lines` lines from fifo `fifo_index`.
+        """Return up to `max_lines` lines from a non blocking fifo.
 
         :return: A list containing the read lines.
         """
         return self.fifo.readlines()[:max_lines]
 
+    def flags(self):
+        """Return flags of the opened fifo.
+
+        :return An integer with flags of the opened file.
+        """
+        fd = self.fifo.fileno()
+        return fcntl.fcntl(fd, fcntl.F_GETFL)
+
+    def or_flags(self, flags):
+        """Or between `flags` arg and opened filed existing flags."""
+        new_flags = self.flags() | flags
+        self.set_flags(new_flags)
+
+    def set_flags(self, flags):
```

Comment:
The getter/setter could be replaced with a `@property` like here: https://github.com/firecracker-microvm/firecracker/blob/eef40ed38718b8ef065a7bf2d53c64b30ffafb62/tests/framework/microvm.py#L169

Reply:
@aghecenco , please take a look! I've added @property decorator, as suggested.

Reply:
The nice thing about this type of decorators is that you can simply do:
```
# getter example
fl  = fifo.flags 
# setter
fifo.flags = OS_NONBLOCK
```
So the perk here is that you can use same name for both getter and setter when decorating them. However, I see you used "set_flags" for the setter.

Reply:
Oh, right. I will push pushed an updated with your usage. Thanks.